ladybird.sh and ladybird.py do not honor CC and CXX flangs

### Summary

When I build ladybird with the following:
`rm -rf Build/vcpkg Build/release && CC=$(which clang) CXX=$(which clang++) ./Meta/ladybird.py run`

ladybird reports that it found the `/usr/bin/c++` compiler instead. 

### Operating system

Linux

### Steps to reproduce

1. Use Fedora 41 and have both gcc and clang compilers
2. Perform a git pull to updat the repo
3. Run the command: `rm -rf Build/vcpkg Build/release && CC=$(which clang) CXX=$(which clang++) ./Meta/ladybird.py run`


### Expected behavior

Program finds the clang compilers

### Actual behavior

system ignores environment variable and uses gcc compilers

```txt
NOTE:
This also leads to ffmpeg failing to build with:
CMake Error at /home/mschramm/Programs/ladybird/Build/release/vcpkg_installed/x64-linux-dynamic/share/vcpkg-pkgconfig-get-modules/x_vcpkg_pkgconfig_get_modules.cmake:31 (execute_process):
  execute_process failed command indexes:

    1: "Abnormal exit with child return code: Segmentation fault"

Call Stack (most recent call first):
  buildtrees/versioning_/versions/ffmpeg/a1c3c785273a5cecf79a78a22ac58c6cb316ac6d/portfile.cmake:870 (x_vcpkg_pkgconfig_get_modules)
  scripts/ports.cmake:206 (include)
```

### URL for a reduced test case

NA

### HTML/SVG/etc. source for a reduced test case

```html
NA
```

### Log output and (if possible) backtrace


```shell
NA
```

### Screenshots or screen recordings

_No response_

### Build flags or config settings

_No response_

### Contribute a patch?

- [ ] I’ll contribute a patch for this myself.

--- a/Meta/host_platform.py
+++ b/Meta/host_platform.py
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: BSD-2-Clause
 
 import enum
+import os
 import platform
 import sys
 
@@ -45,6 +46,11 @@ class Platform:
             sys.exit(1)
 
     def default_compiler(self) -> tuple[str, str]:
+        cc = os.environ.get("CC", None)
+        cxx = os.environ.get("CXX", None)
+
+        if cc and cxx:
+            return (cc, cxx)
         if self.host_system == HostSystem.Windows:
             return ("clang-cl", "clang-cl")
         return ("cc", "c++")

